sos/plugins/etcd.py

```diff
@@ -16,21 +16,39 @@
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 from sos.plugins import Plugin, RedHatPlugin
+import socket
 
 
 class etcd(Plugin, RedHatPlugin):
 
     """etcd plugin
     """
 
+    def port_report(self):
+        """ Until etcd v2, the daemon listens on port 4001 for backward
+        compatibility of v2.0 eariler. But etcd v3 no loger litens on 
+        port 4001 and it listens on port 2379 only. 
+        To collect information in any version, this plugin should check 
+        which port is available.
+        """
+        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        try:
+            self.sock.connect(("localhost", 2379))
+        except socket.error, msg:
+            self.sock.close()
+            return "4001"
+        self.sock.close()
+        return "2379"
+
```

Comment:
I am not at all keen on adding `socket` usage to plugins. This seems like a fragile, and potentially risky way of doing things (we are assuming, based on nothing more than a lack of exception, that whatever socket responds is really `etcd`, and then later we send a bunch of HTTP requests to that port).

Surely this cannot be the way that other components on the system obtain this port value?

Why can't we either parse configuration, version information, or even e.g. `netstat` output and make a better determination?